server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/ReindexingRunningOptionsParser.java

```diff
@@ -28,9 +28,9 @@
 public class ReindexingRunningOptionsParser {
 
     public static RunningOptions parse(Request request) {
-        return intQueryParameter(request, "messagesPerSecond")
-            .map(RunningOptions::new)
-            .orElse(RunningOptions.DEFAULT);
+        return new RunningOptions(
+            intQueryParameter(request, "messagesPerSecond").orElse(RunningOptions.DEFAULT.getMessagesPerSecond()),
+            modeQueryParameter(request, "mode").orElse(RunningOptions.DEFAULT.getMode()));
```

Comment:
We should consider adding a builder to these running options, so that we better handle default values.